File: Observer/Go/observer/observerA.go
package observer

import "fmt"

type ObserverA struct {
	value string
}

func (ob ObserverA) Update(val string) {
	ob.value = val
	ob.Display()
}

func (ob ObserverA) Display() {
	fmt.Printf("observer A value is " + ob.value + "\n")
}

File: Observer/PHP/observerA.php
<?php
  class ObserverA implements Observer
  {
    private $_value;

    public function update($value)
    {
      $this->_value = $value;
      $this->display();
    }

    public function display() {
      echo "Observer A value is " . $this->_value . "\n";
    }
  }
?>
